src/Views/Dock.vala

```diff
@@ -176,6 +161,29 @@ public class Dock : Gtk.Grid {
         attach (pressure_switch, 2, 5, 1, 1);
 
         check_for_screens ();
+
+        switch (dock_preferences.IconSize) {
+            case 48:
+                icon_size_48.active = true;
+                break;
+            case 64:
+                icon_size_64.active = true;
+                break;
+            default:
+                icon_size_32.active = true;
+        }
```

Comment:
I know we don't explicitly support other icon sizes, but this still seems incorrect to assume anything that isn't 48 or 64 is 32. Instead, can't we just not have a selection in that case?

```suggestion
        switch (dock_preferences.IconSize) {
            case 32:
                icon_size_32.active = true;
                break;
            case 48:
                icon_size_48.active = true;
                break;
            case 64:
                icon_size_64.active = true;
                break;
            default:
                debug ("Unsupported icon size");
        }
```

Reply:
Ah I see now since it's a radio, GTK automatically selects the first one anyway if we don't explicitly. Hm…

Reply:
Ah, so throwing something like this in works:

```vala
var icon_size_unsupported = new Gtk.RadioButton.from_widget (icon_size_32);
```

```suggestion
        switch (dock_preferences.IconSize) {
            case 32:
                icon_size_32.active = true;
                break;
            case 48:
                icon_size_48.active = true;
                break;
            case 64:
                icon_size_64.active = true;
                break;
            default:
                icon_size_unsupported.active = true;
                debug ("Unsupported dock icon size);
        }
```

It feels better to me since we're not just selecting the smallest size when they might have some totally different size selected, and it appears to work.

Reply:
@cassidyjames yeah the way we can support that is to have a radio button that we don’t pack into the grid